### PR TITLE
Issue#31/calculator

### PR DESCRIPTION
**Pull request by markiyanshvets**
image

### CODE REVIEW

**Review by oskarpbcc — APPROVED** [review]

**Comment by markiyanshvets** [comment]
Provided a new endpoint.
Endpoint GET /payment configuration has been merged with the payment configuration/exchange endpoint and will be removed 

Updates:
FromCurrency is null => it takes the default value (NOK)
The query is empty => will retrieve each configuration
Markup, PaymentTypes, CountryCode is hidden and details/settings is flatten.

image
image

File: Api/BccPay.Core.Sample/BccPay.Core.Sample.API/Controllers/PaymentConfigurationsController.cs [hunk 1]
```diff
@@ -11,6 +12,13 @@ namespace BccPay.Core.Sample.API.Controllers
     [Route("payment-configurations")]
     public class PaymentConfigurationsController : BaseController
     {
+        private readonly ICurrencyService _exchangeService;
```
Comment (yuriy-butkevych) [inline Api/BccPay.Core.Sample/BccPay.Core.Sample.API/Controllers/PaymentConfigurationsController.cs:15]: Needed?

File: Api/BccPay.Core.Sample/BccPay.Core.Sample.API/Controllers/PaymentConfigurationsController.cs [hunk 2]
```diff
@@ -21,5 +29,17 @@ public async Task<IActionResult> GetPaymentConfigurations([FromQuery] PaymentCon
 
             return Ok(result);
         }
+
+        [HttpGet("with-exchange")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(GetConfigurationsWithExchangedCurrencyQueryResult))]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<IActionResult> CreatePayment([FromQuery] ExchangeWithConfigurationsRequestModel request)
```
Comment (yuriy-butkevych) [inline Api/BccPay.Core.Sample/BccPay.Core.Sample.API/Controllers/PaymentConfigurationsController.cs:36]: rename to `GetConfigurationsWithExchangedCurrency`

File: Api/BccPay.Core.Infrastructure/Helpers/Implementation/CurrencyService.cs [hunk 3]
```diff
@@ -39,43 +39,38 @@ public Task<CurrencyConversionRecord> Exchange(string fromCurrency, string toCur
...
-            decimal exchangeResultNetto = 0;
+            decimal exchangeResult = 0;
+
             if (fromOpposite)
-                exchangeResultNetto = Decimal.Divide(amount, currencyRate);
+                exchangeResult = Decimal.Divide(amount, currencyRate);
             if (!fromOpposite)
-                exchangeResultNetto = Decimal.Multiply(amount, currencyRate);
-
+                exchangeResult = Decimal.Multiply(amount, currencyRate);
```
Comment (yuriy-butkevych) [inline Api/BccPay.Core.Infrastructure/Helpers/Implementation/CurrencyService.cs:62]:
```
decimal exchangeResult = fromOpposite 
    ? Decimal.Divide(amount, currencyRate) 
    : Decimal.Multiply(amount, currencyRate);
```

**Review by yuriy-butkevych — APPROVED** [review]

File: Api/BccPay.Core.Sample/BccPay.Core.Sample.API/Controllers/PaymentConfigurationsController.cs (hunk 2 above)
Comment (markiyanshvets) [reply Api/BccPay.Core.Sample/BccPay.Core.Sample.API/Controllers/PaymentConfigurationsController.cs]: done

File: Api/BccPay.Core.Sample/BccPay.Core.Sample.API/Controllers/PaymentConfigurationsController.cs (hunk 1 above)
Comment (markiyanshvets) [reply Api/BccPay.Core.Sample/BccPay.Core.Sample.API/Controllers/PaymentConfigurationsController.cs]: no

File: Api/BccPay.Core.Infrastructure/Helpers/Implementation/CurrencyService.cs (hunk 3 above)
Comment (markiyanshvets) [reply Api/BccPay.Core.Infrastructure/Helpers/Implementation/CurrencyService.cs]: done

**Review by tkristiansen — APPROVED** [review]

**Pull request merged**